internal/cli/updog.go

```diff
@@ -135,10 +151,16 @@ func provideUpdogSubscriber(objects []client.Object, client client.Client) *updo
 
 func (s *updogSubscriber) SetUp(ctx context.Context, _ store.RStore) error {
 	for _, obj := range s.objects {
+		// Create() modifies its object mysteriously, so copy it first
```

Comment:
Ugh this confused me greatly too at one point - there's weirdly code to restore GVK on `Update()` but not `Create()` and an ominous note on the restore function that it could hopefully go away at some point after deps are upgraded to 1.16, but I think it's past that point already and presumably still an issue 😢 

https://github.com/kubernetes-sigs/controller-runtime/blob/b5065bd85190e92864522fcc85aa4f6a3cce4f82/pkg/client/client.go#L155

Reply:
haha, vincepri has filed some tilt issues, maybe we can trade a tilt fix for fixing that TODO in controller-runtime :joy: